File: NestedTemplate/Samples/java/lib/MyStack.java
package com.stellartech;

import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.Duration;
import software.amazon.awscdk.services.sqs.Queue;
import software.amazon.awscdk.services.sns.Topic;
import software.amazon.awscdk.services.sns.subscriptions.SqsSubscription;
import software.constructs.Construct;

public class NotificationStack extends Stack {
    public NotificationStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        Queue queue = Queue.Builder.create(this, "OrderProcessingQueue")
            .visibilityTimeout(Duration.seconds(300))
            .build();

        Topic topic = Topic.Builder.create(this, "OrderNotificationTopic").build();

        topic.addSubscription(new SqsSubscription(queue));
    }
}

File: NestedTemplate/Samples/javascript/lib/my-stack.js
import { Stack, Duration } from 'aws-cdk-lib';
import { Queue } from 'aws-cdk-lib/aws-sqs';
import { Topic } from 'aws-cdk-lib/aws-sns';
import { SqsSubscription } from 'aws-cdk-lib/aws-sns-subscriptions';

export class InventoryStack extends Stack {
  constructor(scope, id, props) {
    super(scope, id, props);

    const queue = new Queue(this, 'InventoryUpdateQueue', {
      visibilityTimeout: Duration.seconds(300)
    });

    const topic = new Topic(this, 'InventoryNotificationTopic');

    topic.addSubscription(new SqsSubscription(queue));
  }
}